rose/learner.py

```diff
@@ -346,3 +346,159 @@ def _parallel_active_learn():
 
         # block/wait for each workflow until it finishes
         [learner.result() for learner in submitted_learners]
+
+
+class ParallelActiveLearningAlgoSelector(ActiveLearner):
+    """
+    ParallelActiveLearningAlgoSelector is a subclass of ActiveLearner that implements 
+    multiple active learning pipelines in parallel, each pipeline is a 
+    sequential active learning loop, and uses the same simulation and 
+    training tasks, but distinct active learning task.
+    """
+    def __init__(self, engine: ResourceEngine) -> None:
+        ''' 
+        Initialize the ParallelActiveLearningAlgoSelector object.
+
+        Args:
+            engine: The ResourceEngine object that manages the resources and
+            tasks submission to HPC resources during the active learning loop.
+        '''
+        super().__init__(engine, register_and_submit=False)
+        self.active_learn_functions: Dict[str, Dict] = {}
+
+        # A dictionary to store stats for each active learning pipeline
+        # e.g. self._pipeline_stats['algo_1'] = {'iterations': 5, 'last_result': 0.01}
+        self._pipeline_stats: Dict[str, Dict] = {}
+        self.best_pipeline_name = None
+        self.best_pipeline_stats = None
+
+    def _check_stop_criterion_and_return_value(self, stop_task_result):
```

Comment:
This is redundant and almost the same as the base class. As we agreed, once this PR is merged, I will adopt your changes so you can inherit the base class `check_stop_criterion`